### PR TITLE
feat: [CHK-3107] Added creditor reference id to payment notice

### PR DESCRIPTION
**Pull request by petretiandrea**
This PR introduces changed due to WISP deprecation. It allows to save iuv (creditor reference id) to payment notice both in redis and mongodb.

Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/214

#### List of Changes
- added new client id as discriminator for WISP deprecation flow
- added creditor reference id to payment notice object (will be injected by nodo activation's)
- added creditor reference id to payment request info (aka redis cached payment notice's)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/Transaction.java [hunk 1]
```diff
@@ -74,7 +74,11 @@ public enum ClientId {
         /**
          * Transaction originated by IO app
          */
-        IO;
+        IO,
+        /**
+         * Transaction originated by E.C. through cart by WISP discontinuation
+         */
+        WISP;
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/ecommerce/commons/documents/v2/Transaction.java:81]:
```suggestion
        WISP_REDIRECT;
```

File: src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java [hunk 2]
```diff
@@ -1620,4 +1631,105 @@ void canRoundTripQueueTransactionAuthorizationOutcomeWaitingEvent() {
...
+                        it.pagopa.ecommerce.commons.documents.v2.Transaction.ClientId.CHECKOUT,
+                        ID_CART,
+                        PAYMENT_TOKEN_VALIDITY_TIME_SEC,
+                        new EmptyTransactionGatewayActivationData(),
+                        "34a07bc2-4dad-4a9a-a941-86d232829c94"
+                )
+        );
+        event.setId("be09bed4-f0ae-4ef2-8adb-324f720fc702");
+
+        System.out.println(new String(jsonSerializer.serializeToBytes(event), StandardCharsets.UTF_8));
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java:1676]:
```suggestion
```

**Review by pietro-tota — DISMISSED** [review]
Well done, just some trivial issues

File: src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java (hunk 2 above)
Comment (petretiandrea) [reply src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java:1676]: My debug bad

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/Transaction.java (hunk 1 above)
Comment (petretiandrea) [reply src/main/java/it/pagopa/ecommerce/commons/documents/v2/Transaction.java]: Solved by https://github.com/pagopa/pagopa-ecommerce-commons/pull/213/commits/0962abc7c51a504ac7d89ad1f42b43b0f19e7946

File: src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java [hunk 3]
```diff
@@ -1620,4 +1631,105 @@ void canRoundTripQueueTransactionAuthorizationOutcomeWaitingEvent() {
...
```
[the remark this answers is not in the document]
Comment (petretiandrea) [reply src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java]: Solved by https://github.com/pagopa/pagopa-ecommerce-commons/pull/213/commits/d5651eca3c1147cafb97b88ef2389fa209e323b5

File: src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java (hunk 2 above)
Comment (petretiandrea) [reply src/test/java/it/pagopa/ecommerce/commons/documents/v2/serialization/TransactionEventTypeResolverTest.java]: Solved by https://github.com/pagopa/pagopa-ecommerce-commons/commit/d5651eca3c1147cafb97b88ef2389fa209e323b5

**Review by pietro-tota — APPROVED** [review]
🚀

**Pull request merged**